src/modules/ipfs/client.ts

```diff
@@ -0,0 +1,36 @@
+import { Files } from 'formidable';
+import { PinFileResponse } from './types';
+import uploadFile from './nft.storage';
+import { fromDwebLink } from '.';
+
+export default async function UploadFiles(formFiles: Files ) {
+  const files = Object.values(formFiles)
+    .map(maybeFiles =>  maybeFiles instanceof Array ? maybeFiles[0] : maybeFiles)
+
+  const uploadPromises: Promise<any>[] = []
+  files.forEach((file) => {
+    uploadPromises.push(uploadFile(file.path));
+  })
+
+  const results = await Promise.allSettled(uploadPromises)
+
+  const mixedResults: PinFileResponse[] = []
+  await Promise.all(results.map(async(result, index) => {
+    const fileResponse: PinFileResponse = {
+      error: undefined,
+      uri: '',
+      name: files[index].name || '',
+      type: files[index].type || ''
+    }
+    if (result.status === "rejected"){
+      fileResponse.error = result.reason
+    } else {
+      const json = await result.value.json()
```

Comment:
Have some ideas on cleaning this up but involves resolving the json in the `uploadFile`. It should both make the request and parse into json.

Then this can be a transform without promises.